Sources/AWSLambdaEvents/AppSync.swift

```diff
@@ -0,0 +1,129 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2017-2020 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html
+public enum AppSync {
+    public struct Event: Decodable {
+        public let arguments: [String: ArgumentValue]
+
+        public enum ArgumentValue: Codable {
+            case string(String)
+            case dictionary([String: String])
+
+            public init(from decoder: Decoder) throws {
+                let container = try decoder.singleValueContainer()
+                if let strValue = try? container.decode(String.self) {
+                    self = .string(strValue)
+                } else if let dictionaryValue = try? container.decode([String: String].self) {
+                    self = .dictionary(dictionaryValue)
+                } else {
+                    throw DecodingError.dataCorruptedError(in: container, debugDescription: """
+                    Unexpected AppSync argument.
+                    Expected a String or a Dictionary.
+                    """)
+                }
+            }
+
+            public func encode(to encoder: Encoder) throws {
+                var container = encoder.singleValueContainer()
+                switch self {
+                case .dictionary(let array):
+                    try container.encode(array)
+                case .string(let str):
+                    try container.encode(str)
+                }
+            }
+        }
+
+        public let request: Request
+        public struct Request: Decodable {
+            let headers: HTTPHeaders
+        }
+
+        public let source: [String: String]?
+        public let stash: [String: String]?
+
+        public let info: Info
+        public struct Info: Codable {
+            public var selectionSetList: [String]
+            public var selectionSetGraphQL: String
+            public var parentTypeName: String
+            public var fieldName: String
+            public var variables: [String: String]
+        }
+
+        public let identity: Identity?
+        public struct Identity: Codable {
```

Comment:
The documentation in https://docs.aws.amazon.com/appsync/latest/devguide/resolver-context-reference.html#aws-appsync-resolver-context-reference-identity says: 

```
Identity

The identity section contains information about the caller. The shape of this section depends on the authorization type of your AWS AppSync API.

For more information about this section and how it can be used, see Security.

API_KEY authorization
The identity field isn’t populated.

AWS_IAM authorization
The identity has the following shape:

{
    "accountId" : "string",
    "cognitoIdentityPoolId" : "string",
    "cognitoIdentityId" : "string",
    "sourceIp" : ["string"],
    "username" : "string", // IAM user principal
    "userArn" : "string",
    "cognitoIdentityAuthType" : "string", // authenticated/unauthenticated based on the identity type
    "cognitoIdentityAuthProvider" : "string" // the auth provider that was used to obtain the credentials
}

AMAZON_COGNITO_USER_POOLS authorization
The identity has the following shape:

{
    "sub" : "uuid",
    "issuer" : "string",
    "username" : "string"
    "claims" : { ... },
    "sourceIp" : ["x.x.x.x"],
    "defaultAuthStrategy" : "string"
}
```

Should we model this differently?

Reply:
This is a good point! For one thing we need to add the missing fields from AWS_IAM authorization for sure. it would be cool to have an enum that says the type of authorization, but on the other hand this might be fragile as we'd need to base it on the presence of certain fields. Wdyt?